DQM/EcalMonitorTasks/src/GpuTask.cc

```diff
@@ -0,0 +1,455 @@
+      MEs_.erase(std::string("RecHitCpu"));
+      MEs_.erase(std::string("RecHitCpuEnergy"));
+      MEs_.erase(std::string("RecHitCpuTime"));
+      MEs_.erase(std::string("RecHitCpuFlags"));
+      MEs_.erase(std::string("RecHitGpuCpu"));
+      MEs_.erase(std::string("RecHitGpuCpuEnergy"));
+      MEs_.erase(std::string("RecHitGpuCpuTime"));
+      MEs_.erase(std::string("RecHitGpuCpuFlags"));
+    }
+    if (!gpuOnlyPlots_) {
+      MEs_.erase(std::string("DigiGpuAmplitude"));
+      MEs_.erase(std::string("RecHitGpu"));
+      MEs_.erase(std::string("RecHitGpuEnergy"));
+      MEs_.erase(std::string("RecHitGpuTime"));
+      MEs_.erase(std::string("RecHitGpuFlags"));
+      MEs_.erase(std::string("UncalibGpu"));
+      MEs_.erase(std::string("UncalibGpuAmp"));
+      MEs_.erase(std::string("UncalibGpuAmpError"));
+      MEs_.erase(std::string("UncalibGpuPedestal"));
+      MEs_.erase(std::string("UncalibGpuJitter"));
+      MEs_.erase(std::string("UncalibGpuJitterError"));
+      MEs_.erase(std::string("UncalibGpuChi2"));
+      MEs_.erase(std::string("UncalibGpuOOTAmp"));
+      MEs_.erase(std::string("UncalibGpuFlags"));
+    }
+  }
+
+  bool GpuTask::filterRunType(short const* _runType) {
+    for (unsigned iFED(0); iFED != ecaldqm::nDCC; iFED++) {
+      if (_runType[iFED] == EcalDCCHeaderBlock::COSMIC || _runType[iFED] == EcalDCCHeaderBlock::MTCC ||
+          _runType[iFED] == EcalDCCHeaderBlock::COSMICS_GLOBAL ||
+          _runType[iFED] == EcalDCCHeaderBlock::PHYSICS_GLOBAL || _runType[iFED] == EcalDCCHeaderBlock::COSMICS_LOCAL ||
+          _runType[iFED] == EcalDCCHeaderBlock::PHYSICS_LOCAL)
+        return true;
+    }
+
+    return false;
+  }
+
+  void GpuTask::beginEvent(edm::Event const&, edm::EventSetup const&, bool const&, bool&) {
+    EBCpuDigis_ = nullptr;
+    EECpuDigis_ = nullptr;
+    EBCpuUncalibRecHits_ = nullptr;
+    EECpuUncalibRecHits_ = nullptr;
+    EBCpuRecHits_ = nullptr;
+    EECpuRecHits_ = nullptr;
+  }
+
+  template <typename DigiCollection>
+  void GpuTask::runOnCpuDigis(DigiCollection const& _cpuDigis, Collections _collection) {
+    MESet& meDigiCpu(MEs_.at("DigiCpu"));
+    MESet& meDigiCpuAmplitude(MEs_.at("DigiCpuAmplitude"));
+
+    int iSubdet(_collection == kEBCpuDigi ? EcalBarrel : EcalEndcap);
+
+    // Save CpuDigis for comparison with GpuDigis
+    // Static cast to EB/EEDigiCollection during use
+    // Stored as void pointers to make compiler happy
+    if (iSubdet == EcalBarrel)
+      EBCpuDigis_ = &_cpuDigis;
+    else
+      EECpuDigis_ = &_cpuDigis;
+
+    unsigned nCpuDigis(_cpuDigis.size());
+    meDigiCpu.fill(getEcalDQMSetupObjects(), iSubdet, nCpuDigis);
+
+    for (typename DigiCollection::const_iterator cpuItr(_cpuDigis.begin()); cpuItr != _cpuDigis.end(); ++cpuItr) {
+      // EcalDataFrame is not a derived class of edm::DataFrame, but can take edm::DataFrame (digis) in the constructor
+      EcalDataFrame cpuDataFrame(*cpuItr);
```

Comment:
Please use a range for loop:
```suggestion
    for (auto const& digis: _cpuDigis) {
      // EcalDataFrame is not a derived class of edm::DataFrame, but can take edm::DataFrame (digis) in the constructor
      EcalDataFrame cpuDataFrame(digis);
```